jumanji/environments/routing/rware/env.py

```diff
@@ -0,0 +1,529 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+from typing import List, Optional, Sequence, Tuple
+
+import chex
+import jax
+import jax.numpy as jnp
+import matplotlib
+
+from jumanji import specs
+from jumanji.env import Environment
+from jumanji.environments.routing.rware import utils
+from jumanji.environments.routing.rware.generator import Generator, RandomGenerator
+from jumanji.environments.routing.rware.types import (
+    _SHELVES,
+    Action,
+    Agent,
+    Direction,
+    Observation,
+    Shelf,
+    State,
+)
+from jumanji.environments.routing.rware.viewer import RwareViewer
+from jumanji.tree_utils import tree_slice
+from jumanji.types import TimeStep, restart, termination, transition
+from jumanji.viewer import Viewer
+
+
+class Rware(Environment[State]):
```

Comment:
Actually, it seems that the original environment is called `Warehouse` (see [here](https://github.com/semitable/robotic-warehouse/blob/master/rware/warehouse.py#L146)). Should we call our environment the same way? Or call it `RobotWarehouse`? I find `Rware` to be a short for something not super standard.

Reply:
The landing page of the repo calls it RWARE, and so do the related papers that have used it. But happy to call it `RobotWarehouse`, it is more descriptive and it will be clear from the docs that it is a reimplementation of RWARE in JAX. I will make the change.

Reply:
Done